### DISCUSSION TITLE
Trade balloon v2 and v3

### DISCUSSION THREAD

**Discussion by brian-man** [sunflower-land/sunflower-land#2162] [General]
My impression is that devs want to discuss feature proposals here rather than (previously) `-isles-forum` in Discord.  As such, I'm reposting here from
https://discord.com/channels/880987707214544966/1054304170452193330/1054304170452193330

-----

There was some discussion about phasing out the in-game Trade balloon now that NiftySwap is an option.  I hope that idea is rejected as the balloon has at least two significant features that NiftySwap does not:
1. in-game so no withdraw/deposit transaction fees (for buyers and sellers) and no withdraw taxes (for sellers)
2. supported in all markets of the game (Nifty excludes many markets/regions because governments suck)

I'd like to see the following for v2:
1. buyer - allow partial purchase
Every day I want to buy <= 50 of a resource, yet sellers are motivated to list the maximum because of the polygon transaction fee and the game's flat fee.
2. seller - lower listing fee, perhaps %-based
3. seller - increase amount maximums
4. seller - add repricing for cheaper than listing fee
5. seller - add increasing amount button for cheaper than listing fee

I'd like to see the following for v3:
1. make all balloon activity off-chain / SFL server-side
This would remove the blockchain gas cost entirely.  In addition to being nice for sellers, it supports players who want to be more immersed int he game as a game w/o worrying about blockchain more often than during syncs.

**Comment by LandLadyFarmer** [comment]
what they talked about last stream was that nifty or nifty-like trading would be in-games with a different fee system (not 100 if fee was talked on last stream) but definitely in-game trading was talked about how it would work was not mentioned and is needed.

For whats suggested here by brain-man is in all whats needed and have been for a long time and i think most people know it so now when Sunflower isle runs lets do some changes but stop all development on trade balloon....

I would like for trade balloon to be canceled and no further work go in to it rather do a complete rebuild of trading..
Maybe sell direct to pool is not what it should be without additional fees or something but open up for auctions "both blind and open", buy sell orders+ trades like the stock market for those that like more advanced trading and that all fees is based on % of what you sell and buy even if it's needed to be in matic...

I don't think there needs to be an v2 just something new like Trader island, just rebuild it and cancel balloon as soon it's partial ready and advanced trading can come down the line.
Continue to improve trade balloon with fixes or functions 1 by 1 is just to drag out all misery how bad trade balloon is and should not be continued to improve, that would be a lot of work for nothing when a rebuild is ultimate needed!
Have an island you can go to where seller can list trades or do auctions and for buyers to list buy orders so both sellers and buyers can try to make a good deal..

Alt.. Seller can list a high number of resources for buyers to do partial buys like brain-man said but where the seller doesn't have to pay any fee instead the fee gets divided for buyers to pay as tax for not buying bulk...
 
Besides that have some pool system for instant buy and sell but maybe limited on how much you can dump there so all sunflower and potatoes spammer still needs to have an hassle to get rid of their stock or that the fee should significant increase when dumping high numbers and some others limitation so that p2p trades comes first..

Kill trade balloon, open up "Trader island" and have auctions, buy and sell orders, pools, advanced trading, fee based on % of trades, market festivals and a lot more.
No more improvements or partial work on trade balloon, let the balloon take to the sky and go away for ever bye bye balloon have a good journey...


Best regards //Balloonkiller

**Reply by eodestroyer** [reply]
Posted Feb2?  Is this on the list of things to improve?
